test/translator/common_test.cc

```diff
@@ -78,4 +78,40 @@ TEST(Common, ShouldCorrectlyWriteValueToSpan) {
   EXPECT_EQ(0b11000100, buffer[0]);
 }
 
+TEST(Common, ShouldBuildAllocationWithSuccessStatus) {
+  translator::Allocation allocation;
+  auto alloc_callback = [](uint16_t count) -> uint64_t {
+    EXPECT_EQ(1, count);
+    return 1337;
+  };
+  void (*dealloc_callback)(uint64_t, uint16_t) = nullptr;
+  translator::SetAllocPageCallbacks(alloc_callback, dealloc_callback);
+
+  translator::StatusCode status_code = allocation.SetPages(1, 1);
```

Comment:
I would recommend passing and returning different values for data and mdata so that you can verify that the right values made it into each field.

Reply:
That's a good idea, will do!